exception specification for new operator does not match previous declaration

The exception specification of the operator new(size_t) should match the one of the previous declaration. The previous declaration is included from `../../include\new`.

This mismatch is revealed when compiling the vector example with clang (LLVM 11):
```
../common/new.cpp:33:1: error: exception specification in declaration does not match previous declaration
operator new(size_t size) _GLIBCXX_USE_NOEXCEPT
^
../../include\new:73:7: note: previous declaration is here
void* operator new(std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      ^
../common/new.cpp:39:1: error: exception specification in declaration does not match previous declaration
operator new[](size_t size) _GLIBCXX_USE_NOEXCEPT
^
../../include\new:75:7: note: previous declaration is here
void* operator new[](std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      ^
```

--- a/examples/common/new.cpp
+++ b/examples/common/new.cpp
@@ -30,13 +30,13 @@ allocate(size_t size) _GLIBCXX_USE_NOEXCEPT
 }
 
 void *
-operator new(size_t size) _GLIBCXX_USE_NOEXCEPT
+operator new(size_t size)
 {
 	return allocate<true>(size);
 }
 
 void *
-operator new[](size_t size) _GLIBCXX_USE_NOEXCEPT
+operator new[](size_t size)
 {
 	return allocate<true>(size);
 }

